src/fsharp/NicePrint.fs

```diff
@@ -71,6 +71,37 @@ module private PrintUtilities =
         | (x:: rest) -> [ resultFunction x (layoutFunction x -- leftL (match rest.Length with 1 -> FSComp.SR.nicePrintOtherOverloads1() | n -> FSComp.SR.nicePrintOtherOverloadsN(n))) ] 
         | _ -> []
 
+    /// Layout a reference to a type 
+    let layoutTyconRefImpl isAttribute (denv: DisplayEnv) (tcref:TyconRef) = 
+        let demangled = 
+            let name =
+                if denv.includeStaticParametersInTypeNames then 
+                    tcref.DisplayNameWithStaticParameters 
+                elif tcref.DisplayName = tcref.DisplayNameWithStaticParameters then
+                    tcref.DisplayName // has no static params
+                else
+                    tcref.DisplayName+"<...>" // shorten
+            if isAttribute then 
+                defaultArg (String.tryDropSuffix name "Attribute") name 
+            else name
+        let tyconTextL = wordL demangled
+        if denv.shortTypeNames then 
+            tyconTextL
+        else
+            let path = demangledPathOfCompPath tcref.CompilationPath
+            let path =
```

Comment:
@dungpa I have not seen that code was moved only.
btw @dsyme about shadowing (asking, like for `<|` on code review ) is it usually ok?